chirp/drivers/icr6.py

```diff
@@ -0,0 +1,794 @@
+"""Icom IC-R6 Driver"""
+# Copyright 2024 John Bradshaw Mi0SYN <john@johnbradshaw.org>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+# TODO:
+# Channel: Frequency setting
+# Channel: TSQL freq or DTCS code with polarity
+# Channel: Skip scan settings
+# Channel: Comment (not sure if supported in ICF)
+# Banks - channel assignment and name editing
+# Validate TS by freq/band (e.g. 9k TS is only for AM brdcast. Airband?)
+#   CS-R6 shows red X if setting invalid, use that to verify Chirp-output ICF
+# Scan edges per manual
+# Radio settings - lamp, delays, keytones, ...
+
+# Notes:
+#
+# USA models have WX alert and block certain frequencies (see manual).
+
+from chirp.drivers import icf
+from chirp import chirp_common, directory, bitwise
+from chirp.settings import RadioSettingGroup, RadioSetting, \
+    RadioSettingValueBoolean, RadioSettingValueList, \
+    RadioSettingValueString, \
+    RadioSettingValueFloat, RadioSettings
+#    RadioSettingValueInteger, \
+
+mem_format = """
+// Channel memories: 1300x 16-byte blocks
+// 0x000 to 0x513F inclusive
+struct {
+  u8 freq0;               // Freq: low byte
+  u8 freq1;               // Freq: mid byte
+  u8 freq_flags:6,        // Related to multiplier (step size)
+     freq2:2;             // Freq: high bits - 18 bits total = step count
+  u8 af_filter:1,         // AF Filter: 0=Off, 1=On
+     attenuator:1,        // Attenuator: 0=Off, 1=On
+     mode:2,              // Modulation: Index to "MODES"
+     tuning_step:4;       // Freq tuning steps: Index to "STEPS"
+  u8 unknown4ab:2,
+     duplex:2,            // 0 = None, 1 = Minus, 2 = Plus
+     unknown4e:1,
+     tmode:3;             // TSQL/DTCS Setting: Index to "TONE_MODES"
+  u8 offset_l;            // Offset - low value byte
+  u8 offset_h;            // Offset - high value byte
+  u8 unknown7:2,
+     ctone:6;             // TSQL Index. Valid range: 0 to 49
+  u8 unknown8;
+  u8 canceller_freq_h;    // Canceller training freq index - high 8 bits
+  u8 canceller_freq_l:1,  //                               - LSB
+     unknown10m:4,
+     vsc:1,               // Voice Squelch Control: 0=Off, 1=On
+     canceller:2;         // USA-only Canceller option: Index to "CANCELLER"
+  u8 name0;
+  u8 name1;
+  u8 name2;
+  u8 name3;
+  u8 name4;
+} memory[1300];
+
+// Unknown: 0x5140 to 0x5F7F inclusive
+struct {
```

Comment:
Theres no need to do this sort of thing and it just makes this longer and less readable. Use `#seek nbytes` or `#seekto addr` to skip a section you're not going to handle.

Reply:
Fixed